intermediate_source/realtime_rpi.rst

```diff
@@ -0,0 +1,259 @@
+
+Installation will take at least a few minutes depending on your internet speed and sdcard speed. Once it's done it should look like:
+
+.. image:: https://user-images.githubusercontent.com/909104/152867425-c005cff0-5f3f-47f1-922d-e0bbb541cd25.png
+
+Time to put your sdcard in your Raspberry Pi, connect the camera and boot it up.
+
+.. image:: https://user-images.githubusercontent.com/909104/152869862-c239c980-b089-4bd5-84eb-0a1e5cf22df2.png
+
+
+Once that boots and you complete the initial setup you'll need to edit the ``/boot/config.txt`` file to enable the camera.
+
+.. code:: toml
+
+    # This enables the extended features such as the camera.
+    start_x=1
+
+    # This needs to be at least 128M for the camera processing, if it's bigger you can just leave it as is.
+    gpu_mem=128
+
+    # You need to commment/remove the existing camera_auto_detect line since this causes issues with OpenCV/V4L2 capture.
+    #camera_auto_detect=1
+
+And then reboot. After you reboot the video4linux2 device ``/dev/video0`` should exist.
+
+Installing PyTorch and OpenCV
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+PyTorch and all the other libraries we need have ARM 64-bit/aarch64 variants so you can just install them via pip and have it work like any other Linux system.
+
+.. code:: shell
+
+    $ pip install torch torchvision torchaudio
+    $ pip install opencv-contrib-python
+    $ pip install numpy --upgrade
+
+.. image:: https://user-images.githubusercontent.com/909104/152874260-95a7a8bd-0f9b-438a-9c0b-5b67729e233f.png
+
+
+We can now check that everything installed correctly:
+
+.. code:: shell
+
+  $ python3 -c "import torch; print(torch.__version__)"
+  1.10.0+cpu
+
+.. image:: https://user-images.githubusercontent.com/909104/152874271-d7057c2d-80fd-4761-aed4-df6c8b7aa99f.png
+
+
+Video Capture
+~~~~~~~~~~~~~~
+
+For video capture we're going to be using OpenCV to stream the video frames
+instead of the more common ``picamera``. `picamera` isn't available on 64-bit
+Raspberry Pi OS and it's much slower than OpenCV. OpenCV directly accesses the
+``/dev/video0`` device to grab frames.
+
+The model we're using (MobileNetV2) takes in image sizes of ``224x224`` so we
+can request that directly from OpenCV at 36fps. We're targeting 30fps for the
+model but we request a slightly higher framerate than that so there's always
+enough frames.
+
+.. code:: python
+
+  import cv2
+  from PIL import Image
+
+  cap = cv2.VideoCapture(0)
+  cap.set(cv2.CAP_PROP_FRAME_WIDTH, 224)
+  cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 224)
+  cap.set(cv2.CAP_PROP_FPS, 36)
+
+OpenCV returns a ``numpy`` array in BGR so we need to read and do a bit of
+shuffling to get it into the expected RGB format.
+
+.. code:: python
+
+    ret, image = cap.read()
+    # convert opencv output from BGR to RGB
+    image = image[:, :, [2, 1, 0]]
+
+NOTE: You can get even more performance by training the model directly with OpenCV's BGR data format to remove the conversion step.
```

Comment:
you can link to a doc here

Reply:
I don't really have any docs here, this would be on the training dataset.